src/react/components/pages/modelCompose/modelCompose.tsx

```diff
@@ -367,7 +367,7 @@ export default class ModelComposePage extends React.Component<IModelComposePageP
         this.composeModalRef.current.open([], false, false);
 
         if (model.attributes.isComposed) {
-            const apiVersion = this.props.project?.apiVersion || constants.apiVersion;
+            const apiVersion = (constants.enableAPIVersionSelection && this.props.project?.apiVersion) ? this.props.project.apiVersion : constants.apiVersion;
```

Comment:
Does "this.props.project?.apiVersion" always have value? If its value comes from the apiVersionPicker, it won't be empty, right? In that case we don't need to check  constants.enableAPIVersionSelection either.

Reply:
It **doesn't** always have a value. Older projects created before we added API version selection, have an undefined value. So, it's for backwards compatibility with older projects